pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -202,41 +234,35 @@ func (c *EtcdCertSignerController) sync(ctx context.Context, syncCtx factory.Syn
 	return updateErr
 }
 
-func (c *EtcdCertSignerController) syncAllMasterCertificates(ctx context.Context, recorder events.Recorder) error {
-	// TODO(thomas): it is of utmost importance to keep the existing signer certs for now
-	// when we just create a new signer cert, the new revision does not allow the peer to join the existing two-node
-	// cluster based on the old CA. Any newly rotated additions will come for free through the signerCaBundle and will work out of the box.
-	// Unfortunately, we can't make use of the certrotation.RotatedSigningCASecret here because it would immediately try to rotate the existing signer.
-	signerCaPair, err := tlshelpers.ReadConfigSignerCert(ctx, c.secretClient)
-	if err != nil {
-		return err
-	}
-
-	c.reportExpirationMetric(signerCaPair, "signer-ca")
-
-	// TODO(thomas): we need to transition that new signer as a replacement for the above - today we only bundle it
-	newSignerCaPair, _, err := c.certConfig.signerCert.EnsureSigningCertKeyPair(ctx)
```

Comment:
Please bear with me as I try to walk through the rollout sequence to make sure I have this right :) and cut me off if I'm assuming something incorrect.

### The old way:
We had `openshift-config/etcd-signer` secret as the `old-CA` and `openshift-etcd/etcd-signer` secret as the `new-CA` both of which were bundled on all etcd pods but all server certs were only signed by the `old-CA`. 
And we updated and overwrote `openshift-etcd/etcd-signer`=>`openshift-config/etcd-signer` to regenerate the leaf certs. That way the first pod that rolled out a the new revision (with the `new-CA` signed server cert) would still be recognized by the older revision etcd members (since they already had the `new-CA` trust bundled)

### The new way:
With this change we only have `openshift-etcd/etcd-signer` as the definitive CA secret right?
So call that the `old-CA`.
If we now delete `openshift-etcd/etcd-signer` and have it get recreated with `new-CA` (e.g as Revision 2), then we won't regenerate the leaf certs until we first have all etcd pods bundled with `new-CA` at Revision 2 right?

So at revision 2 we have all etcd members trust bundled via `etcd-all-bundles` with only `new-CA` then.
But aren't the server certs still signed by the `old-CA` that is no longer in the trust bundle anymore?

So IIUC it's effectively `--cert-file=<old-CA signed server.crt>` and `--trusted-ca-file=<new-CA.crt>`.

But seems like this is working in CI so trying to figure out what I'm missing in how the cluster is working before the leaf certs are re-generated.

Reply:
> So at revision 2 we have all etcd members trust bundled via etcd-all-bundles with only new-CA then.
> But aren't the server certs still signed by the old-CA that is no longer in the trust bundle anymore?

They are! otherwise we would've split the cluster, the bundle itself is stateful, so if we present it a new CA it would just append it. Revoking the old CA would be either a manual step, or another control loop to delete the configmap once in a while.